src/core/endpoints/push/remove_push_channels.js

```diff
@@ -8,18 +8,27 @@ export function getOperation(): string {
 }
 
 export function validateParams(modules: ModulesInject, incomingParams: ModifyDeviceArgs) {
-  let { device, pushGateway, channels } = incomingParams;
+  let { device, pushGateway, channels, topic } = incomingParams;
   let { config } = modules;
 
   if (!device) return 'Missing Device ID (device)';
-  if (!pushGateway) return 'Missing GW Type (pushGateway: gcm or apns)';
+  if (!pushGateway) return 'Missing GW Type (pushGateway: gcm, apns or apns2)';
   if (!channels || channels.length === 0) return 'Missing Channels';
   if (!config.subscribeKey) return 'Missing Subscribe Key';
+
+  if (pushGateway === 'apns2') {
```

Comment:
Why not do =>
```
if ((pushGateway === 'apns2') && !topic) return 'Missing APNS2 topic';
```
?

Reply:
Fixed  in 0699a066298c3e101e0ec6ca41b41cf0bed4a40c